src/aws/osml/model_runner/inference/http_detector.py

```diff
@@ -0,0 +1,109 @@
+import logging
+from io import BufferedReader
+from json import JSONDecodeError
+from typing import Optional
+
+import geojson
+import urllib3
+from aws_embedded_metrics.logger.metrics_logger import MetricsLogger
+from aws_embedded_metrics.metric_scope import metric_scope
+from aws_embedded_metrics.unit import Unit
+from geojson import FeatureCollection
+
+from aws.osml.model_runner.api import ModelInvokeMode
+from aws.osml.model_runner.app_config import MetricLabels, ServiceConfig
+from aws.osml.model_runner.common import Timer
+
+from .detector import Detector
+from .endpoint_builder import FeatureEndpointBuilder
+from .feature_utils import create_mock_feature_collection
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPDetector(Detector):
+    def __init__(self, endpoint: str) -> None:
+        """
+        A HTTP model endpoint invoking object, intended to query sagemaker endpoints.
+
+        :param endpoint: str = the full URL to invoke the model
+
+        :return: None
+        """
+        self.http_pool = urllib3.PoolManager(cert_reqs="CERT_NONE")
```

Comment:
How are timeouts of the HTTP endpoint addressed here? I'm not sure but guessing that setting the pool manager up this way is going to use a system default timeout which might not be ideal. Consider setting up this pool with an actual timeout value and maybe allow that to be passed as part of the configuration for these HTTP endpoints.

Reply:
Implemented backoff policy for the pool manager.